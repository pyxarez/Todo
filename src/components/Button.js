import React from 'react';

import './Button.res/style.css'

Button.propTypes = {
  type: React.PropTypes.string,
  onClick: React.PropTypes.func,
}

export default function Button({type, onClick, value}) {
  return type
    ? <button
        className={`my-button-component button-${type}`}
        onClick={onClick}>{value}</button>
    : <button
        className={'my-button-component'}
        onClick={onClick}>{value}</button>
}
