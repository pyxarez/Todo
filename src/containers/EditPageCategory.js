import React, { Component, PropTypes } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { browserHistory } from 'react-router';

import { changeTaskLocation } from '../store/actions/CategoryActions';

import Button from '../components/Button';
import Category from '../components/Category';

import ShareExtendState from './ShareExtendState';

import './CategoryContainer.res/style.css';

export class Container extends Component {
  static propTypes = {
    id: PropTypes.node.isRequired,
    title: PropTypes.string.isRequired,
    extended: PropTypes.bool.isRequired,
    onExtend: PropTypes.func.isRequired,
    changeTaskLocation: PropTypes.func.isRequired,
    getChildrenCount: PropTypes.func.isRequired,
    currentCategoryId: PropTypes.number.isRequired,
    currentTaskId: PropTypes.number.isRequired,
    children: React.PropTypes.oneOfType(
      [
        React.PropTypes.arrayOf(React.PropTypes.element.isRequired),
        React.PropTypes.bool,
        React.PropTypes.object
      ]
    ),
  }

  handleChangeTaskLocation = () => {
    const {
      id,
      currentCategoryId,
      currentTaskId,
      changeTaskLocation
    } = this.props;

    changeTaskLocation(currentCategoryId, id, currentTaskId);
    browserHistory.goBack();
  }

  render() {
    const {
      id,
      title,
      extended,
      onExtend,
      getChildrenCount,
      children,
      currentCategoryId
    } = this.props;

    return (
      <div className='my-categoryContainer-component'>
        <Category
          id={id}
          title={title}
          onExtend={ getChildrenCount(children) > 0 && onExtend }>
          {currentCategoryId !== id &&
            <Button
              type="swap"
              onClick={this.handleChangeTaskLocation} />}
        </Category>
        {children &&
          <div className={extended
            ? "categories categories_extended"
            : "categories categories_hidden"}>{children}</div>}
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    changeTaskLocation: bindActionCreators(changeTaskLocation, dispatch),
  }
}

const EditPageCategoryContainer = ShareExtendState(Container);
export default connect(null, mapDispatchToProps)(EditPageCategoryContainer)
