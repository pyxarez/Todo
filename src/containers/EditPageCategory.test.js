import React from 'react';
import { shallow } from 'enzyme';

import { Container } from './EditPageCategory';
import Button from '../components/Button';

const setup = () => {
  const props = {
    id: 0,
    title: 'Vassa',
    extended: true,
    onExtend: () => {},
    changeTaskLocation: jest.fn(),
    getChildrenCount: jest.fn(() => 1),
    currentCategoryId: 1,
    currentTaskId: 2
  };
  const wrapper = shallow(
    <Container {...props}>
      <div className='vassa'></div>
    </Container>
  );

  return {
    wrapper,
    props
  }
}

describe('EditCategoryContainer', () => {
  it('should render self and subconmponents', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render self and subconmponents \
    without children and button', () => {
    const { props } = setup();
    const newProps = {
      ...props,
      getChildrenCount: jest.fn(() => 0),
      currentCategoryId: 0,
      id: 0
    };
    const wrapper = shallow(<Container {...newProps} />)

    expect(wrapper).toMatchSnapshot();
  });

  it('should call changeTaskLocation', () => {
    const { wrapper, props } = setup();
    expect(wrapper).toMatchSnapshot();

    const buttonComponent = wrapper.find(Button);
    buttonComponent.simulate('click');

    expect(props.changeTaskLocation)
      .toBeCalledWith(
        props.currentCategoryId,
        props.id,
        props.currentTaskId
      );
  });

  it('should render self and subconmponents \
  with not extended container', () => {
    const { props } = setup();
    const newProps = {
      ...props,
      extended: false
    };
    const wrapper = shallow(
      <Container {...newProps} >
        <div></div>
      </Container>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
