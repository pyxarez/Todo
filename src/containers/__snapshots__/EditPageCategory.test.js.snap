exports[`EditCategoryContainer should call changeTaskLocation 1`] = `
<div
  className="my-categoryContainer-component">
  <Category
    id={0}
    onExtend={[Function]}
    title="Vassa">
    <Button
      onClick={[Function]}
      type="swap" />
  </Category>
  <div
    className="categories categories_extended">
    <div
      className="vassa" />
  </div>
</div>
`;

exports[`EditCategoryContainer should render self and subconmponents     without children and button 1`] = `
<div
  className="my-categoryContainer-component">
  <Category
    id={0}
    onExtend={false}
    title="Vassa" />
</div>
`;

exports[`EditCategoryContainer should render self and subconmponents   with not extended container 1`] = `
<div
  className="my-categoryContainer-component">
  <Category
    id={0}
    onExtend={[Function]}
    title="Vassa">
    <Button
      onClick={[Function]}
      type="swap" />
  </Category>
  <div
    className="categories categories_hidden">
    <div />
  </div>
</div>
`;

exports[`EditCategoryContainer should render self and subconmponents 1`] = `
<div
  className="my-categoryContainer-component">
  <Category
    id={0}
    onExtend={[Function]}
    title="Vassa">
    <Button
      onClick={[Function]}
      type="swap" />
  </Category>
  <div
    className="categories categories_extended">
    <div
      className="vassa" />
  </div>
</div>
`;
